hawkular/alerts.py

```diff
@@ -0,0 +1,159 @@
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+"""
+
+from __future__ import unicode_literals
+import client
+
+try:
+    # Python 3
+    from urllib.parse import quote, urlencode
+except ImportError:
+    # Fall back to Python 2's urllib2
+    from urllib import quote, urlencode
+
+try:
+    import simplejson as json
+except ImportError:
+    import json
+
+
+class ApiJsonEncoder(json.JSONEncoder):
+    def default(self, obj):
+        if isinstance(obj, ApiOject):
+            return obj.to_json_object()
+        else:
+            return json.JSONEncoder.default(self, obj)
+
+
+class ApiOject:
+    def to_api_object(self, dict):
+        pass
+
+    def to_json_object(self):
+        return self.transform_dict_to_camelcase(self.__dict__)
+
+    def _to_camelcase(self, word):
+        s = ''.join(x.capitalize() or '_' for x in word.split('_'))
+        return ''.join([s[0].lower(), s[1:]])
+
+    def _to_underscore(self, word):
+        return ''.join(["_" + c.lower() if c.isupper() else c for c in word]).strip('_')
+
+    def transform_dict_to_camelcase(self, dictionary):
+        return dict((self._to_camelcase(k), v) for k, v in dictionary.iteritems() if v is not None)
+
+    def transform_dict_to_underscore(self, dictionary):
+        return dict((self._to_underscore(k), v) for k, v in dictionary.iteritems() if v is not None)
+
+    @classmethod
+    def transform_to_object(cls, o):
+        if isinstance(o, cls):
+            return o
+        elif isinstance(o, dict):
+            return cls(o)
+        elif isinstance(o, list):
+            return [cls(ob) for ob in o]
+        else:
+            return None
+
+
+class Trigger(ApiOject):
+    def __init__(self, dictionary=dict()):
+        self.id = None
+        self.name = None
+        self.description = None
+        self.type = None
+        self.event_type = None
+        self.event_category = None
+        self.event_text = None
+        self.severity = None
+        self.context = None
+        self.tags = None
+        self.actions = None
+        self.auto_disable = None
+        self.auto_enable = None
+        self.auto_resolve = None
+        self.auto_resolve_alerts = None
+        self.auto_resolve_match = None
+        self.data_id_map = None
+        self.member_of = None
+        self.enabled = None
+        self.firing_match = None
+        self.source = None
+        self.__dict__.update(self.transform_dict_to_underscore(dictionary))
```

Comment:
@simon3z 

I would prefer to initialize all variables to None because it helps me when I use autocomplete on my IDE, it makes any sense for you? Also, I think is a good practice. IMHO.

But I agree about filtering the hash values.

WDYT?